foolscap/connection_plugins.py

```diff
@@ -52,3 +55,26 @@ def hint_to_endpoint(self, hint, reactor):
             raise InvalidHintError("unrecognized TCP hint")
         host, port = mo.group(1), int(mo.group(2))
         return endpoints.TCP4ClientEndpoint(reactor, host, port), host
+
+def default_tcp4_endpoint_generator(*args, **kw):
+    """
+    Default generator used to create client-side TCP4ClientEndpoint
+    instances.  We do this to make the unit tests work...
+    """
+    return endpoints.TCP4ClientEndpoint(*args, **kw)
+
+@implementer(IConnectionHintHandler)
+class SocksPlugin:
+    def __init__(self, socks_host, socks_port, proxy_endpoint_generator=default_tcp4_endpoint_generator):
```

Comment:
You can replace `default_tcp4_endpoint_generator ` with `endpoints.TCP4ClientEndpoint` and remove the `default_tcp4_endpoint_generator` function above.